### PR TITLE
Added example region-tag for Family.js

### LINKED ISSUE #157: @example Comments should refer to sample code
nodejs-bigtable has `@example` comments that include code.  This is error prone, since it's not compiled and may get obsolete.  Instead of inline code, all `@example`s should refer to a sample file and a tag within it.

nodejs-spanner has a great examples of this.  Here's one example ([link](https://github.com/googleapis/nodejs-spanner/blob/72efac10bcab961732cca234312908fa1b5bc3dd/src/batch-transaction.js#L116)):

```
/**
 ...
 * @example <caption>include:samples/batch.js</caption>
 * region_tag:spanner_batch_client
*/
```

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes #157 for family.js (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/document-snippets/tests/family.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    });
+    await instance.createTable(TABLE_ID);
+  });
+
+  after(async () => {
+    await instance.delete();
+  });
+
+  it('should create a column family', function(done) {
+    familySnippets.createColmFamily(INSTANCE_ID, TABLE_ID, FAMILY_ID, err => {
```
Comment (stephenplusplus) [inline samples/document-snippets/tests/family.js]: Just passing `done` in place of the `err => { assert.ifError(err); done(); }` is equivalent! `done` will either report the error it received, or end the test successfully if it didn't receive an error.

File: samples/document-snippets/family.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    const instance = bigtable.instance(instanceId);
+    const table = instance.table(tableId);
+    const family = table.family(familyId);
+    // [START bigtable_create_family]
+    family
+      .create()
+      .then(result => {
+        const family = result[0];
+        // let apiResponse = result[1];
+        callback(null, family);
```
Comment (stephenplusplus) [inline samples/document-snippets/family.js]: Will this line be shown in the docs? If so, it could be hard to follow that we're calling the promise-style API, but passing the results back to a `callback` that doesn't seem to be defined.

Comment (vijay-qlogic) [reply samples/document-snippets/family.js]: @stephenplusplus 
Yes this line will be shown in docs. 
You are correct the callback in docs wont make clear understanding
'll update the code with callback outside the region-tags

File: samples/document-snippets/family.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    const family = table.family(familyId);
+    // [START bigtable_get_family]
+    family
+      .get()
+      .then(result => {
+        const family = result[0];
+        // const apiResponse = result[1];
+      })
+      .catch(err => {
+        // handle error
```
Comment (stephenplusplus) [inline samples/document-snippets/family.js]: "Error handling omitted." or "Handle the error." would match our phrasing conventions better.

File: samples/document-snippets/family.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        const metaData = result[0];
+        // const apiResponse = result[1];
+      })
+      .catch(err => {
+        // handle error
+      });
+    // [END bigtable_get_family_meta]
+    callback();
+  },
+  setMetaData: (instanceId, tableId, familyId, callback) => {
```
Comment (stephenplusplus) [inline samples/document-snippets/family.js]: `setMetadata`

File: samples/document-snippets/family.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        const family = result[0];
+        // const apiResponse = result[1];
+      })
+      .catch(err => {
+        // handle error
+      });
+    // [END bigtable_get_family]
+    callback();
+  },
+  getMetaData: (instanceId, tableId, familyId, callback) => {
```
Comment (stephenplusplus) [inline samples/document-snippets/family.js]: `getMetadata`

File: samples/document-snippets/family.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+      });
+    // [END bigtable_get_family_meta]
+    callback();
+  },
+  setMetaData: (instanceId, tableId, familyId, callback) => {
+    const instance = bigtable.instance(instanceId);
+    const table = instance.table(tableId);
+    const family = table.family(familyId);
+    // [START bigtable_set_family_meta]
+    var metadata = {
```
Comment (stephenplusplus) [inline samples/document-snippets/family.js]: Can we go `const`?

File: samples/document-snippets/tests/family.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (stephenplusplus) [inline samples/document-snippets/tests/family.js]: 2018*

File: samples/document-snippets/tests/family.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+  it('should check family exists', function(done) {
+    familySnippets.existsFamily(INSTANCE_ID, TABLE_ID, FAMILY_ID, done);
+  });
+
+  it('should get the family', function(done) {
+    familySnippets.getFamily(INSTANCE_ID, TABLE_ID, FAMILY_ID, done);
+  });
+
+  it('should get family meta-data', function(done) {
```
Comment (stephenplusplus) [inline samples/document-snippets/tests/family.js]: `metadata`

File: samples/document-snippets/tests/family.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    familySnippets.getMetaData(INSTANCE_ID, TABLE_ID, FAMILY_ID, done);
+  });
+
+  it('should set family meta-data', function(done) {
```
Comment (stephenplusplus) [inline samples/document-snippets/tests/family.js]: `metadata`

File: samples/document-snippets/family.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+ */
+
+const snippets = {
+  createColmFamily: (instanceId, tableId, familyId) => {
+    const Bigtable = require('@google-cloud/bigtable');
+    const bigtable = new Bigtable();
+    const instance = bigtable.instance(instanceId);
+    const table = instance.table(tableId);
+    const family = table.family(familyId);
+    // [START bigtable_create_family]
```
Comment (sduskis) [inline samples/document-snippets/family.js]: Please move the `// [START` ... comment to line 18.

File: samples/document-snippets/family.js [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    // [END bigtable_create_family]
+  },
+  existsFamily: (instanceId, tableId, familyId) => {
+    const Bigtable = require('@google-cloud/bigtable');
+    const bigtable = new Bigtable();
+    const instance = bigtable.instance(instanceId);
+    const table = instance.table(tableId);
+    const family = table.family(familyId);
+
+    // [START bigtable_exists_family]
```
Comment (sduskis) [inline samples/document-snippets/family.js]: Please move up the tag.

**Comment by sduskis** [comment]
@vijay-qlogic, can you please address the comments?

**Comment by sduskis** [comment]
@stephenplusplus, can we move forward with this PR?
